CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+    # add the install directory (`dist`) to CMAKE_PREFIX_PATH.
+
+    find_package(OpenAssetIO REQUIRED)
+endif ()
+
+# For Windows tests to work we must add the dlls to the PATH
+if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+    file(TO_NATIVE_PATH "${OpenAssetIO_BINARY_DIR}" OpenAssetIO_BINARY_DIR_NATIVE)
+endif ()
+
+#-----------------------------------------------------------------------
+# C++ tests.
+
+add_executable(test.cpp.core src/test.core.cpp)
+add_test(cpp.core test.cpp.core)
+target_link_libraries(test.cpp.core PRIVATE OpenAssetIO::openassetio-core)
+
+
+# GCC's C++11 ABI toggle must match OpenAssetIO's settings.
+if (DEFINED OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+    if (OPENASSETIOTEST_GLIBCXX_USE_CXX11_ABI)
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=1)
+    else ()
+        target_compile_definitions(test.cpp.core PRIVATE _GLIBCXX_USE_CXX11_ABI=0)
+    endif ()
+endif ()
+
+if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+    set_tests_properties(cpp.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+endif ()
+
+#-----------------------------------------------------------------------
+# C tests.
+
+if (OPENASSETIOTEST_ENABLE_C)
+    add_executable(test.c.core src/test.core.c)
+    add_test(c.core test.c.core)
+    target_link_libraries(test.c.core PRIVATE OpenAssetIO::openassetio-core-c)
+    if (WIN32 AND DEFINED OpenAssetIO_BINARY_DIR)
+        set_tests_properties(c.core PROPERTIES ENVIRONMENT "PATH=${OpenAssetIO_BINARY_DIR_NATIVE}")
+    endif ()
+endif ()
+
+#-----------------------------------------------------------------------
+# Python tests.
+
+if (OPENASSETIOTEST_ENABLE_PYTHON)
+
+    if (OPENASSETIOTEST_ENABLE_SUBPROJECT)
+        message(
+            FATAL_ERROR
+            "Testing OpenAssetIO as a subproject when Python is enabled is currently unsupported."
+            " The binary Python module must be built and made available alongside the Python"
+            " sources for tests to work."
+        )
+    endif ()
+
+    find_package(Python COMPONENTS Interpreter Development.Embed REQUIRED)
+
+    if (NOT "${Python_VERSION}" STREQUAL "${OpenAssetIO_Python_VERSION}")
+        message(
+            FATAL_ERROR
+            "Found Python '${Python_VERSION}' but OpenAssetIO was built against"
+            " '${OpenAssetIO_Python_VERSION}'"
+        )
+    endif ()
+
+    #-----------------------------------------------------------------------
+    # Python sources tests.
+
+    # Python sources test.
+    add_test(NAME python COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/src/test.py)
+    set_tests_properties(python PROPERTIES ENVIRONMENT PYTHONPATH=${OpenAssetIO_Python_SITELIB})
+
+    #-----------------------------------------------------------------------
+    # Python bridge test (loading Python plugin system from C++).
+
+    add_executable(test.python.bridge src/test.python.bridge.cpp)
+    add_test(python.cpp.bridge test.python.bridge)
```

Comment:
Nit just to say, my eyes went to 🍝 by reading the different permutations of python, bridge, cpp used in these two lines

Reply:
:grin: Yeah it's tricky, I didn't want to use the word "test" in what is already obviously a test, and didn't want to use the word "cpp" in what is obviously a `.cpp` file, but then also wanted to capture those things in the source file and test target name, respectively.

Reply:
Might be easier if python bridge and cpp were always in the same order?

Reply:
Yeah OK if that reads easier.  I had in my mind "the Python-to-C++ bridge" so `python.cpp.bridge` kinda worked, but `python.bridge.cpp` works too.